### ISSUE TITLE
document how to reload network interface naming when systemd-networkd is no used

### ISSUE THREAD

**Issue by calestyo** [systemd/systemd#26601]
### Component

other

### Is your feature request related to a problem? Please describe

I'm **not** using `systemd-networkd` but have a `/etc/systemd/network/10-eth0.link` like:
```
[Match]
MACAddress=FA:16:3E:77:9E:83

[Link]
Name=eth0
```
in order to get the interface named the way I like it.

When setting an image up the first time, the MAC isn't set (correctly) so I get the default iface naming schema.

Even after quite intensive searching I found no way (that really always works) that allows to rename the interfaces according to these config files, while the system is running.<br>
Some resources on the web said that:
```
udevadm trigger --verbose --subsystem-match=net --action=add
```
would do this, but that doesn't always seem to work.<br>
So right now I simply always reboot.

`networkctl(1)` mentions the `reconfigure` action, but that anyway works only when `systemd-networkd` is used.

Could you please document (I guess in `systemd.network(5)`) how this can be done?

Tanks,
Chris.

### The systemd version you checked that didn't have the feature you are asking for

252.5

**Comment by yuwata** [comment]
Renaming to the name used by kernel (ethX, wlanX, and so on) is not supported, and easily causes race. Please use different name for the interface. Closing.

Note, using `udevadm trigger` is the correct way to reconfigure interface name.

**Comment by calestyo** [comment]
@yuwata 

Well then same point with using names other than the kernel names.

And as mentioned before `udevadm trigger` doesn't work.

Just tried it again on a 6.1.12 kernel, systemd/udev 252.6.

Changed `/etc/systemd/network/10-eth0.link` to:
```
[Match]
MACAddress=FA:16:3E:AE:62:31

[Link]
Name=foo0
```
then
```
# udevadm trigger
```

but not message in the kernel log about any renaming and `ip link show` also continues to display the old name.

**Comment by yuwata** [comment]
The comment in my comment is abbreviated, you need to call it with the add action as you said in the first comment:
```
# udevadm trigger --verbose --subsystem-match=net --action=add
```

**Comment by calestyo** [comment]
Again, as written in my first post (where I gave exactly that command),... doesn't work.

Perhaps the reason is what's described here: https://superuser.com/questions/914646/renaming-network-interface-with-systemd/1326238#1326238

Anyway, I think the following works (at least for me):
- the iface must be DOWN
- `systemctl restart systemd-udev-trigger.service`

**Comment by yuwata** [comment]
> - the iface must be DOWN

Yes. This is also a crucial point. Then, calling `udevadm trigger ...` manually should also work.

**Comment by yuwata** [comment]
@calestyo I added a short explanation in systemd.link(5) in #26668. If you'd like, please take a look.

Tentatively reopen this page, and will close this again after the PR being merged.

**Comment by calestyo** [comment]
> Yes. This is also a crucial point. Then, calling udevadm trigger ... manually should also work.

I think that I used to have cases where even with the iface being DOWN, it didn't work - which is why I wrote that it wouldn't have worked sometimes. Any I'm 99% sure that I had it down when I tried when commenting here earlier.

Anyway,... right now it always succeeds, so maybe I'm just wrong and it wasn't down.

> I added a short explanation in systemd.link(5) in #26668. If you'd like, please take a look.

Looks good. The only minor thing that could have been made was indicating the `eth0` is the old name. But in principle this should be clear from the context... so fine by me as it is.

Thanks.
